modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java

```diff
@@ -1103,6 +1110,23 @@ private IgniteCheckedException shutdown(CacheStoreHolder holder, boolean cleanFi
         return aggr;
     }
 
+    private void cleanParentDirectories(CacheStoreHolder holder) {
+        Arrays.stream(holder.partStores).map(store -> ((FilePageStoreV2)store).getPath().getParent()).distinct().forEach(new Consumer<Path>() {
```

Comment:
Doing such class casts is a bad habit, maybe we should introduce new method to interface